### PR TITLE
feat: add homeView > section query field

### PR DESCRIPTION
**Pull request by dblandin**
**Example**

```graphql
{
  homeView {
    section(id: "home-view-section-auction-lots-for-you") {
      __typename
      ... on GenericHomeViewSection {
        component {
          title
        }
      }
    }
  }
}
```

### CODE REVIEW

File: src/schema/v2/homeView/getSectionsForUser.ts [hunk 1]
```diff
@@ -22,65 +23,19 @@ export async function getSectionsForUser(
...
+      RecentlyViewedArtworks,
     ]
   }
 
   return sections
 }
-
-// stub sections
-
-type HomeViewSection = {
```
Comment (MounirDhahri) [inline src/schema/v2/homeView/getSectionsForUser.ts:44]: nice 👏

**Review by MounirDhahri — APPROVED** [review]

File: src/schema/v2/homeView/index.ts [hunk 2]
```diff
@@ -34,13 +40,35 @@ const SectionConnection: GraphQLFieldConfig<any, ResolverContext> = {
...
+      description: "The ID of the section",
+      type: new GraphQLNonNull(GraphQLString),
+    },
+  },
+  resolve: (_root, { id }, context) => {
+    const { meLoader } = context.authenticatedLoaders
+
+    if (!meLoader) throw new Error("You must be signed in to see this content.")
+
+    if (id.length === 0) {
```
Comment (MounirDhahri) [inline src/schema/v2/homeView/index.ts:57]:
```suggestion
    if (!id) {
```

**Comment by MounirDhahri** [comment]
I will merge this to avoid copy pasting the GraphQL schema and to also in order to merge Nikita's Eigen PR

**Comment by MounirDhahri** [comment]
There is nothing wrong with this PR but something to keep in mind - whenever we're going to make a breaking change, we will break the push schema step in Eigen since we run `yarn relay` in CI. It should be fine though because I am updating Eigen currently with the new schema

Screenshot 2024-08-01 at 15 29 59

File: src/schema/v2/homeView/index.ts (hunk 2 above)
Comment (dblandin) [reply src/schema/v2/homeView/index.ts:57]: The argument is already enforced by the schema to be non-null. I think this check (which I borrowed from a similar resolver) guards against an empty string.

**Comment by dblandin** [comment]
> It should be fine though because I am updating Eigen currently with the new schema

Yep, good to know. I was (and still am) assuming that we can follow-up quickly with [this PR](https://github.com/artsy/eigen/pull/10524) to resolve the breaking changes. I'll touch base in that PR.

**Pull request merged**